pkg/controller/controller_pod.go

```diff
@@ -135,7 +136,7 @@ func (tc *TFJobController) reconcilePods(
 	}
 
 	// Update the active status since we have created -diff pods during the loop.
-	tfjob.Status.TFReplicaStatuses[rtype].Active = int32(len(activePods) - diff)
+	tfjob.Status.TFReplicaStatuses[rtype].Active = expected
```

Comment:
What if expected is negative?

Reply:
I think it will never happen sine succeeded pods always < replicas, WDYT

Reply:
How about use `activePods` directly to make the code more readability?

Reply:
activePods is outdated since we have already created or deleted some pods during the loop

Reply:
But creation maybe failed so we should only trust the active pods we get in this loop.

Reply:
I think if the creation is failed, the functions returns errors and the control flow can not reach here.

Reply:
I mean creation API is return ok, but `Pod` are not running successfully because some scheduling or resource limit error, in another word `Pod` are pending not active.

Reply:
In our definition pending pods is also active pods :thinking: 

```
func IsPodActive(p *v1.Pod) bool {
	return v1.PodSucceeded != p.Status.Phase &&
		v1.PodFailed != p.Status.Phase &&
		p.DeletionTimestamp == nil
}
```

Reply:
I opened a new issue #484

Reply:
SGTM.